src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -678,6 +686,9 @@ public NumberType getNumberType() throws IOException
         if ((_numTypesValid & NR_BIGDECIMAL) != 0) {
             return NumberType.BIG_DECIMAL;
         }
+        if ((_numTypesValid & NR_FLOAT) != 0) {
```

Comment:
I have some concerns wrt backwards-compatibility here but...  I guess there's already dependency wrt calls made that affect type so maybe this is fine.